spock-spring/src/main/java/org/spockframework/spring/xml/MockBeanDefinitionParser.java

```diff
@@ -28,10 +29,18 @@ protected String getBeanClassName(Element element) {
   @Override
   protected void doParse(Element element, BeanDefinitionBuilder builder) {
 
+    if (element.getLocalName().equals("spy") && element.getAttribute("ref").isEmpty()
+          && element.getAttribute("class").isEmpty()) {
+      throw new BeanCreationException(element.getAttribute("id"),
+             "Cannot create spy. Please specify either 'class' or 'ref' attributes.");
+    }
     // configure MockFactory
     builder.addConstructorArgValue(element.getAttribute("class"));
     builder.addPropertyValue("name", element.getAttribute("id"));
     builder.addPropertyValue("mockNature", element.getLocalName());
+    if(!element.getAttribute("ref").isEmpty()) {
```

Comment:
Shouldn't it be either `ref` or `class`, what happens if the `class` does not match the actual class of the bean?